src/css/css.tsx

```diff
@@ -592,6 +592,57 @@ const parseProps = {
  * @author strarsis <strarsis@gmail.com>
  * @author modified by: msand <msand@abo.fi>
  */
+
+function extractVariables(stylesheet: CssNode): Map<string, string> {
+  const variables = new Map<string, string>();
+
+  csstree.walk(stylesheet, {
+    visit: 'Declaration',
+    enter(node) {
+      const { property, value } = node as Declaration;
+      if (property.startsWith('--')) {
+        const variableName = property.trim();
+        const variableValue = csstree.generate(value).trim();
+        variables.set(variableName, variableValue);
+      }
+    },
+  });
+
+  return variables;
+}
+
+function resolveVariables(
+  value: string | CssNode | undefined,
+  variables: Map<string, string>
+): string {
+  if (value === undefined) {
+    return '';
+  }
+  const valueStr = typeof value === 'string' ? value : csstree.generate(value);
+  return valueStr.replace(
+    /var\((--[^,)]+)(?:,\s*([^)]+))?\)/g,
+    (_, variableName, fallback) => {
+      const resolvedValue = variables.get(variableName);
+      if (resolvedValue !== undefined) {
+        return resolveVariables(resolvedValue, variables);
+      }
+      return fallback ? resolveVariables(fallback, variables) : '';
+    }
+  );
+}
+
+const propsToResolve = ['fill', 'color', 'stroke', 'stopColor', 'floodColor'];
```

Comment:
As per w3 specs:
`currentColor` applies to properties `fill`, `stroke`, `stop-color`, `flood-color` and `lighting-color` to I guess we should add `lightingColor` here

Reply:
https://www.w3.org/TR/SVG11/color.html
https://www.w3.org/TR/SVG11/filters.html#AnExample
```
<svg width="7.5cm" height="5cm" viewBox="0 0 200 120">
  <style>
    body {
      --my-color: #00f;
    }
  </style>
  <filter id="MyFilter" filterUnits="userSpaceOnUse" x="0" y="0" width="200" height="120">
    <feGaussianBlur in="SourceAlpha" stdDeviation="4" result="blur"/>
    <feOffset in="blur" dx="4" dy="4" result="offsetBlur"/>
    <feSpecularLighting in="blur" surfaceScale="5" specularConstant=".75" specularExponent="20" lighting-color="var(--my-color)" result="specOut">
      <fePointLight x="-5000" y="-10000" z="20000"/>
    </feSpecularLighting>
    <feComposite in="specOut" in2="SourceAlpha" operator="in" result="specOut"/>
    <feComposite in="SourceGraphic" in2="specOut" operator="arithmetic" k1="0" k2="1" k3="1" k4="0" result="litPaint"/>
    <feMerge>
      <feMergeNode in="offsetBlur"/>
      <feMergeNode in="litPaint"/>
    </feMerge>
  </filter>
  <rect x="1" y="1" width="198" height="118" fill="#888888" stroke="blue"/>
  <g filter="url(#MyFilter)">
    <g>
      <path fill="none" stroke="#D90000" stroke-width="10" d="M50,90 C0,90 0,30 50,30 L150,30 C200,30 200,90 150,90 z"/>
      <path fill="#D90000" d="M60,80 C30,80 30,40 60,40 L140,40 C170,40 170,80 140,80 z"/>
      <g fill="#FFFFFF" stroke="black" font-size="45" font-family="Verdana">
        <text x="52" y="76">SVG</text>
      </g>
    </g>
  </g>
</svg>
```